app/forms/registration_form.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class RegistrationForm < ApplicationForm
+  def save(create_params)
+    return false unless client_valid?(create_params[:client_id])
+
+    user = User.new(create_params.except(:client_id))
+
+    if user.invalid?
+      user_errors = user.errors
+      errors.add(:user, user_errors.full_messages.last) unless user_errors.empty?
```

Comment:
`last` is a bit confusing here, we can use `to_sentence` instead.  And given the user is `invalid?` then i think we don't need to check the `user.errors` is `empty?` or not anymore.

```suggestion
      errors.add(:user, user.errors.full_messages.to_sentence)
```